components/movementsensor/gpsutils/cachedData.go

```diff
@@ -224,11 +231,15 @@ func (g *CachedData) ReadSatsInView(ctx context.Context) (int, error) {
 	return g.nmeaData.SatsInView, nil
 }
 
-// ReadSatsInUse returns the number of satellites in use to calculate fix type.
-func (g *CachedData) ReadSatsInUse(ctx context.Context) (int, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	return g.nmeaData.SatsInUse, nil
+// GetCommonReadings returns a map including fix value, sats in view and sats in use.
+func (g *CachedData) GetCommonReadings(ctx context.Context) (map[int]interface{}, error) {
+	commonReadings := make(map[int]interface{})
+
+	commonReadings[FixValueKey] = g.nmeaData.FixQuality
```

Comment:
`g.nmeaData.FixQuality` might be mutated from a different thread while we're trying to read it, causing trouble. Try guarding it by locking (and deferring unlocking) the mutex at the top of the function.